guides-markdown/multisig-wallet.md

```diff
@@ -0,0 +1,359 @@
+> This could be usefull when you accidently registered a wrong key.
+> But if it has reached to `n` , you must create a new account.
+
+Finally, do the same thing for Bob. Then the multisig account is ready.
+
+### 4. Receive incoming information from the wallet server.
+
+Ok, so let's check the wallet's status by
+
+```sh
+curl ${url}/wallet/sharedWallet1/account/multisig1?token=${sharedWalletToken}
+```
+
+You should see something like
+
+```json
+{
+  "name": "multisig1",
+  "initialized": true,
+  "witness": true,
+  "watchOnly": true,
+  "type": "multisig",
+  "m": 2,
+  "n": 3,
+  "accountIndex": 1,
+  "receiveDepth": 1,
+  "changeDepth": 1,
+  "nestedDepth": 1,
+  "lookahead": 10,
+  "receiveAddress": "tb1..." , // bech32 address for receiving
+  "changeAddress": "tb1...", // address for change
+  "nestedAddress": "2M...", // base58check address for P2SH-nested-P2WSH,
+  "accountKey": "tpub...", // tpubs you've registered above
+  "keys": [
+    "tpub...",
+    "tpub..."
+  ],
+  "balance": {
+    "tx": 0,
+    "coin": 0,
+    "unconfirmed": 0,
+    "confirmed": 0
+  }
+}
+```
+
+You can see that values in a `balance` are all zero.
+So before going to next section, please send the fund to the account's address.
+
+But before sending, let's listen on to the wallet with websocket interface
+
+```sh
+bwallet-cli --id=sharedWallet1 \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${sharedWalletToken} \
+  listen
+```
+
+Sending can be done in anyway you like.
+Check out our [TX creation guide](http://bcoin.io/guides/working-with-txs.html) if you wan't to do it using bcoin api.
+
+Ready? than let's continue
+
+### 5. Spending funds from the wallet.
+
+TODO:
+
+### 6. Recovering the wallet.
+
+Having private key in local itself doesn't mean that your funds are safe.
+even when the node is compromised or deleted completely, you must be able to recover it.
+So let's assume that `sharedWallet1` has been deleted completely.
+
+Recovering procedure is quite simple. You just repeat the public key registration process as we did above, and run `rescan` comamnd at the end.
+
+```sh
+bwallet-cli \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \ # Must use your admin token to create a new wallet
+  --watch-only=true \ # make it watch only
+  --type=multisig \
+  --m=2 \
+  --n=3 \
+  --name=multisig1 \
+  mkwallet sharedWallet2
+
+# add shared xpub(tpub) again. You must repeat this `n` times for all keys
+bwallet-cli \
+  --id=sharedWallet2 \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \
+  --acount=multisig1 \
+  shared add <tpub...>
+
+# and then rescan
+bwallet-cli \
+  --id=sharedWallet2 \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \ # rescanning requires admin token
+  --acount=multisig1 \
+  rescan 10000  # start rescanning from height 10000
```

Comment:
`rescan` is slightly problematic at this time.

See https://github.com/bcoin-org/bcoin/pull/532

It really only works on a full archival node. SPV users should probably use `reset` and pruned full-node users might be out of luck if there are transactions on-chain before their prune height. Might be good to add a warning here.

Reply:
:+1: